pages/api/openPR.js

```diff
@@ -152,63 +133,73 @@ function nomineeSubmission(values, sortedSubmission) {
     organizations: values.organizations ? [values.organizations] : [],
     stage: values.stage ? values.stage : "",
   };
-
   // Order nominee fields in the correct order e.g organizations
   sortedSubmission = orderFields(sortedSubmission);
   return sortedSubmission;
 }
 
+function submitPullRequests(projectName, filesObject) {
+  const MyOctokit = Octokit.plugin(createPullRequest);
+  const octokit = new MyOctokit({
+    auth: TOKEN,
+  });
+
+  // initialize an array for Promises
+  let promiseArray = [];
+
+  for (const submission in filesObject) {
+    let singleFile = {};
+    singleFile[submission.toString()] = filesObject[submission.toString()];
+
+    // Returns a normal Octokit PR response
+    // See https://octokit.github.io/rest.js/#octokit-routes-pulls-create
+    promiseArray.push(
+      composeCreatePullRequest(octokit, {
+        owner: GITHUB_OWNER,
+        repo: GITHUB_REPO,
+        title: `Add nominee: ${projectName}`,
+        body:
+          "Automatic addition of a new nominee submitted through the online form available at https://digitalpublicgoods.net/submission",
+        base: GITHUB_BRANCH,
+        head: createGithubCheckoutBranch(projectName),
+        changes: [
+          {
+            // optional: if `files` is not passed, an empty commit is created instead
+            files: singleFile,
+            commit: `BLD: Add ${projectName}`,
+          },
+        ],
+      })
+    );
+  }
+  // return the array of promises
+  return promiseArray;
+}
+
 export default async (req, res) => {
   if (req.method === "POST") {
     let values = req.body.values;
-
     let nomineeJSON, sdgNumber, evidenceText, sortedSubmission;
 
     // Exclude contact information from pull request
     delete values["contact"];
 
     // Parse SDG information
     values = getSDGRelevanceInfo(values, sdgNumber, evidenceText);
-
     sortedSubmission = nomineeSubmission(values, sortedSubmission);
-
     // Convert JavaScript submission sorted object into JSON string and add newline at EOF
     nomineeJSON = JSON.stringify(sortedSubmission, null, 2).concat("\n");
 
-    const MyOctokit = Octokit.plugin(createPullRequest);
-
-    const octokit = new MyOctokit({
-      auth: TOKEN,
+    let result = await Promise.allSettled(
+      submitPullRequests(getProjectName(values), getSubmissionFiles(values, nomineeJSON))
+    ).then((results) => {
+      console.log("The final result ", results);
+      // There's the opportunity to check if any of them failed but that's for another PR
+      let finalResult = results[0].value;
+      finalResult.number = results[0].value.data.number;
```

Comment:
This should account for the fact that the function may return one or two numbers, so the type that this function returns should be an array, which will contain either one or two elements. This change should be propagated to the front end to display one or two links [here](https://github.com/lacabra/submission-digitalpublicgoods/blob/379d01033e29fdc83eaf67997abf262b8dc60e54/pages/thank-you.js#L75). It gets a bit tricky because we are passing values through a GET parameters to the page.... the call is initiated [here](https://github.com/lacabra/submission-digitalpublicgoods/blob/379d01033e29fdc83eaf67997abf262b8dc60e54/pages/form.js#L167)

Reply:
Thanks for pointing me straight to the files to work on.

Reply:
I'm done with all the requested changes @lacabra